src/manual/using/integration/frontend.md

```diff
@@ -32,9 +32,292 @@ Taskcluster API and refresh when necessary.
+to our action link and status container:
+
+```html
+<script>
+'use strict';
+
+const link = document.getElementById('action-link');
+const status = document.getElementById('status');
+</script>
+```
+
+Now we are going to create an instance of a `WebAuth` from auth0. This is
+entity that lets us communicate with the the auth0 service. This `WebAuth`
+instance will take some properties based on how your auth0 client is configured.
+
+```js
+const link = document.getElementById('action-link');
+const status = document.getElementById('status');
+const auth = new auth0.WebAuth({
+  domain: 'auth.mozilla.auth0.com',
+  responseType: 'token id_token',
+  scope: 'full-user-credentials openid profile',
+  clientID: '<YOUR AUTH0 CLIENT ID>',
+  audience: '<YOUR AUTH0 CLIENT AUDIENCE>',
+  // This redirect URI is configured with your auth0 client.
+  // For this example, we will pretend that this has been configured
+  // to http://localhost:5050/login
+  redirectUri: '<YOUR AUTH0 CLIENT REDIRECT URL>'
+});
+```
+
+With our auth0 client instance we can now respond to actions. Let's capture
+clicks on our action link:
+
+```js
+// We are going to get an access token from auth0
+// which we can give to Taskcluster. Let's store
+// that here when we get it
+let accessToken;
+
+link.addEventListener('click', (e) => {
+  // make sure the page doesn't redirect on click
+  e.preventDefault();
+  
+  // If we don't yet have out access token, let's go
+  // get it from auth0 by having the user sign in
+  if (!accessToken) {
+    return auth.authorize();
+  }
+});
+```
+
+Loading our page up to this point you will see our action link, and clicking
+it will pop up a page to sign in. Once the sign in is complete, auth0 will
+redirect back to the `redirectUri` you specified to `WebAuth` with credentials
+contained in the URL hash. Using `parseHash` from `WebAuth`, we can parse this
+into a friendlier format. Let's add this login after our event handler.
+
+```js
+// Let's parse credentials from the hash
+// if the hash contains data
+if (window.location.hash) {
+  auth.parseHash((err, result) => {
+    // If there was a problem logging in,
+    // let's show it to the user in our status container
+    if (err) {
+      status.innerText = err.errorDescription;
+    } else if (!result || !result.idTokenPayload) {
+      // If we didn't have an auth result, something went wrong.
+      // At a minimum, let's inform the user about the problem.
+      status.innerText = 'Authentication is missing payload';
+    } else {
+      // The user signed in successfully!
+      // Let's save the accessToken, update the UI,
+      // and tell the user
+      accessToken = result.accessToken;
+      status.innerText = 'Authorization successful';
+      link.innerText = 'Get Taskcluster scopes';
+    }
+  });
+}
+```
+
+Great, now with our user logged in, and an access token to identify them,
+we can now authenticate with Taskcluster. In the previous step we
```

Comment:
"we can now call Taskcluster API methods.  We'll call [Auth.currentScopes](https://docs.taskcluster.net/reference/platform/taskcluster-auth/references/api#currentScopes) as an example."